larq_compute_engine/mlir/python/end2end_test.py

```diff
@@ -0,0 +1,79 @@
+import sys
+import pytest
+import larq as lq
+import subprocess
+import tensorflow as tf
+import numpy as np
+from tensorflow.python.eager import context
+
+from larq_compute_engine.mlir.python.converter import convert_keras_model
+
+
+@pytest.fixture
+def eager_mode():
+    """pytest fixture for running test in eager mode"""
+    with context.eager_mode():
+        yield
+
+
+def testnet():
+    img_input = tf.keras.layers.Input(shape=(4, 4, 1))
+    # A layer with zero-padding
+    out = lq.layers.QuantConv2D(
+        filters=32,
+        kernel_size=3,
+        padding="same",
+        input_quantizer="ste_sign",
+        kernel_quantizer="ste_sign",
+        use_bias=False,
+        activation="relu",
+    )(img_input)
+    out = tf.keras.layers.BatchNormalization(
+        gamma_initializer=tf.keras.initializers.RandomNormal(1.0),
+        beta_initializer="uniform",
+    )(out)
+    # A layer with one-padding
+    out = lq.layers.QuantConv2D(
+        filters=16,
+        kernel_size=3,
+        padding="same",
+        # pad_values=1.0,
+        input_quantizer="ste_sign",
+        kernel_quantizer="ste_sign",
+        use_bias=False,
+        activation="relu",
+    )(out)
+    out = tf.keras.layers.BatchNormalization(
+        gamma_initializer=tf.keras.initializers.RandomNormal(1.0),
+        beta_initializer="uniform",
+    )(out)
+    out = tf.keras.layers.GlobalAvgPool2D()(out)
+    return tf.keras.Model(inputs=img_input, outputs=out)
+
+
+def test_simple_model(eager_mode, tmp_path):
+    model = testnet()
+    converted = convert_keras_model(model)
+    modelfile = tmp_path / "end2end_model.tflite"
+    with modelfile.open("wb") as f:
+        f.write(converted)
+
+    x = np.zeros(shape=(1, 4, 4, 1))
```

Comment:
Should we feed in random data here, or would this be tricky with the C++ side?

Reply:
Yes we can do that. Write the random data to a txt file and read that in C++ again. I'll try to do that.